src/storage/storage.ts

```diff
@@ -126,3 +133,30 @@ export class Storage {
     return this.appInternal;
   }
 }
+
+class FirebaseStorageFile extends File{
+  async getDownloadUrl(): Promise<string> {
+    const [metadata] = await this.getMetadata();
+    if (!metadata?.metadata?.firebaseStorageDownloadTokens) {
+      throw new FirebaseError({
+        code: 'storage/no-download-token',
+        message: 'No download token available. Please create one in the Firebase Console.',
```

Comment:
we actually do have a token API that the console uses. We would have to implement it here as well but perhaps we could point users to that method instead of the console